frontend/src/features/admin-form/responses/ResponsesPage/storage/useDecryptionWorkers.ts

```diff
@@ -181,7 +190,24 @@ const useDecryptionWorkers = ({
                       errorCount++
                       console.error('Error in getResponseInstance', e)
                     }
+
                     if (downloadAttachments && decryptResult.downloadBlob) {
+                      // Ensure attachments downloads are spaced out to avoid browser blocking downloads
+                      if (progress % ATTACHMENT_DOWNLOAD_CONVOY_SIZE === 0) {
+                        const now = new Date().getTime()
+                        const elaspedSinceXDownloads =
+                          now - timeSinceLastXAttachmentDownload
+
+                        const waitTime = Math.max(
+                          0,
+                          ATTACHMENT_DOWNLOAD_CONVOY_MINIMUM_SEPARATION_TIME -
+                            elaspedSinceXDownloads,
+                        )
+                        if (waitTime >= 0) {
+                          await waitForMs(waitTime)
+                        }
```

Comment:
Is this strict equality? Cuz no point awaiting if we can download now

Reply:
Good catch!

Reply:
[f95a0c1](https://github.com/opengovsg/FormSG/pull/7449/commits/f95a0c1ff384a4ac954220bd4ea1159363dd4045)